src/out.rs

```diff
@@ -21,42 +19,28 @@ fn main() {
     }
 }
 
-/// Launch a buffer which captures samples piped via stdin then sends them to the audio stream
-/// which should output to the default audio device.
 fn run<T>(device: &cpal::Device, config: &cpal::StreamConfig) -> Result<(), anyhow::Error>
 where
     T: cpal::Sample,
 {
-    // Some setup
-    let stdin = io::stdin();
+    // The number of output channels
     let channels = config.channels as usize;
 
-    // Channel for sending samples from the buffer to the output stream.
-    let (buffer_tx, buffer_rx) = unbounded();
-
-    // Closure for retreiving a single sample from the buffer
-    let mut next_value = move || -> (f64, f64) {
-        let sample = buffer_rx
-            .recv()
-            .expect("receive sample from buffer channel");
+    // Closure for retreiving a single sample from sdtin
+    let mut next_output_sample = move || -> (f64, f64) {
+        // If there is no sample we create some silence.
```

Comment:
I was accounting for a case where a module outputs samples slower than they are being played (< 44100 p/s). In this situation we need something for the output otherwise we would get dropouts. 

That's my logic anyway, may not be 100% solid ;-p

Reply:
It may be that the `cpal` (audio streaming lib) already accounts for this...).

Reply:
I don't see how the code catches that case though. `utils::next_sample` returns `None` if and only if stdin closed and has no idea about a sample rate (or the concept of a "second" for that matter).

Reply:
Oh yeh, this was a brain fart on my part. I was assuming we would get `None` from the iterator when it was "empty" (no sample from stdin), but that's not the behaviour... 

Will fix it up 👍